LiteCore/tests/LazyVectorAPITest.cc

```diff
@@ -0,0 +1,354 @@
+            closeDB();
+            db         = REQUIRED(c4db_openNamed(name, &config, ERROR_INFO()));
+            _wordsColl = Retained(REQUIRED(db->getCollection({"words", "_default"})));
+        }
+        {  // Create encoded target vector parameter
+            Encoder enc;
+            enc.beginDict();
+            enc.writeKey("target");
+            enc.writeData(slice(kWordsTargetVector, 300 * sizeof(float)));
+            enc.endDict();
+            _encodedTarget = enc.finish();
+        }
+    }
+
+    using UpdaterFn = function_ref<bool(LazyIndexUpdate*, size_t, fleece::Value)>;
+
+    static bool alwaysUpdate(LazyIndexUpdate*, size_t, fleece::Value) { return true; }
+
+    /// Get the LazyIndex with the given name. Will return null if the index does not exist.
+    [[nodiscard]] Retained<LazyIndex> getLazyIndex(std::string_view name) const noexcept {
+        auto& store = asInternal(db)->dataFile()->defaultKeyStore();
+        try {
+            return make_retained<LazyIndex>(store, name);
+        } catch ( [[maybe_unused]] std::exception& e ) { return nullptr; }
+    }
+
+    void checkQueryReturnsWords(C4Query* query, const std::vector<string>& expectedWords) const {
+        auto e = REQUIRED(c4query_run(query, _encodedTarget, ERROR_INFO()));
+        REQUIRE(c4queryenum_getRowCount(e, ERROR_INFO()) == expectedWords.size());
+        for ( size_t i = 0; i < expectedWords.size(); ++i ) {
+            INFO("i=" << i);
+            REQUIRE(c4queryenum_next(e, ERROR_INFO()));
+            FLArrayIterator columns  = e->columns;
+            slice           word     = Value(FLArrayIterator_GetValueAt(&columns, 0)).asString();
+            float           distance = Value(FLArrayIterator_GetValueAt(&columns, 1)).asFloat();
+            Log("%.*s: %.3f", FMTSLICE(word), distance);
+            CHECK(word == slice(expectedWords[i]));
+            // CHECK(fabs(distance - expectedDistances[i]) < 0.01);
+        }
+        CHECK(!c4queryenum_next(e, ERROR_INFO()));
+        Log("done");
+        c4queryenum_release(e);
+    }
+
+    void checkQueryReturnsVectors(C4Query* query, int64_t expectedRowCount,
+                                  const std::vector<float>& expectedVectors) const {
+        auto e = REQUIRED(c4query_run(query, _encodedTarget, ERROR_INFO()));
+        REQUIRE(c4queryenum_getRowCount(e, ERROR_INFO()) == expectedRowCount);
+        for ( size_t i = 0; i < expectedRowCount; ++i ) {
+            INFO("i=" << i);
+            REQUIRE(c4queryenum_next(e, ERROR_INFO()));
+            FLArrayIterator columns     = e->columns;
+            auto            vectorArray = Value(FLArrayIterator_GetValueAt(&columns, 0)).asArray();
+            for ( size_t j = 0; j < expectedVectors.size(); j++ ) {
+                float vector = vectorArray[j].asFloat();
+                CHECK(vector == expectedVectors[j]);
+            }
+        }
+        CHECK(!c4queryenum_next(e, ERROR_INFO()));
+        Log("done");
```

Comment:
Is this intended? It's kind of weird in the log when seeing only "done".

Reply:
I copied that from the `LazyVectorQueryTest`. Can remove it.